youtube_dl/extractor/xtube.py

```diff
@@ -51,38 +50,62 @@ def _real_extract(self, url):
         req.add_header('Cookie', 'age_verified=1; cookiesAccepted=1')
         webpage = self._download_webpage(req, display_id)
 
-        flashvars = self._parse_json(
-            self._search_regex(
-                r'xt\.playerOps\s*=\s*({.+?});', webpage, 'player ops'),
-            video_id)['flashvars']
+        jsonstr = self._search_regex(
+                r'MHP1138\.createPlayer\(\'xtubePlayer\',\s*{.*medias:\s*({.+?}),\s*menu:',
+                webpage, 'player ops', flags=re.DOTALL, group=1)
+
+        # Need to fix JSON because Python doesn't like it as it is
+        # This code is very simple and only works with what the site returns
+        # Will probably break if the site changes the output at all...
+        # adds double quotes around key names
+        # changes single quotes in to double quotes around string values
+        newjsonstr = ""
+        for line in jsonstr.split('\n'):
+            line = line.strip()
+            if line[0:2] == '//':
+                continue
+            elif line[0].isalpha(): # key
+                newjsonstr += '\"' + line[0:line.index(':')] + '\":'
+                if line[line.index(':') + 2] == '\'': # string value
+                    newjsonstr += ' \"' + line[line.index(':') + 3:line.rindex('\'')]\
+                                  + '\",'
+                else:
+                    newjsonstr += line[line.index(':') + 1:]
+            else:
+                newjsonstr += line
```

Comment:
It mostly works, but in this case, the site outputs a single line comment in the JSON (`//`) which _parse_json() isn't OK with.